enumerable_methods.rb

```diff
@@ -0,0 +1,100 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i]
+    end
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    length.times do |i|
+      yield to_a[i], i if block_given?
+    end
+  end
+
+  def my_select
+    return enum_for(:select) unless block_given?
+
+    new_array = []
+    my_each do |value|
+      new_array << value if yield value
+    end
+    new_array
+  end
+
+  def my_all?(arg = nil)
+    is_all_true = true
+    my_each do |value|
+      block_given? ? (is_all_true = false unless yield value) : (is_all_true = false unless value == arg)
+    end
+    is_all_true
+  end
```

Comment:
`my_all?` should handle three cases. 
1. When a block is given
2. When a pattern is given and
3. When no block or pattern is given

2 and 3 are not properly handled. See examples below:

```ruby
# 2. When a pattern is given

[1, 2i, 3.14].all?(Numeric)  
#=> true

[1, 2i, 3.14].my_all?(Numeric)  
#=> false

# 3. When no block or pattern is given

[1,2].all?
#=> true

[1,2].my_all?
#=> false
```
HINT
Using the `===` we can achieve this.

```ruby
      if block_given? 
        is_all_true = false unless yield value
      elsif arg.nil?
        is_all_true = false unless value
      else
        is_all_true = false unless arg === value
      end
```

This suggestion applies to `my_any?` and `my_none?` too.

Please check https://ruby-doc.org/core-2.6.5/Enumerable.html for more test cases and general behaviour of these methods.